x-pack/plugins/security_solution/public/cases/links.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { getCasesDeepLinks } from '@kbn/cases-plugin/public';
+import { CASES_PATH, SecurityPageName } from '../../common/constants';
+import { FEATURE, LinkItem } from '../common/links/types';
+
+export const getCasesLinkItems = (): LinkItem => {
+  const casesLinks = getCasesDeepLinks<LinkItem>({
+    basePath: CASES_PATH,
+    extend: {
+      [SecurityPageName.case]: {
+        globalNavEnabled: true,
+        globalNavOrder: 9006,
+        features: [FEATURE.casesRead],
+      },
+      [SecurityPageName.caseConfigure]: {
+        features: [FEATURE.casesCrud],
+        isPremium: true,
+      },
+      [SecurityPageName.caseCreate]: {
+        features: [FEATURE.casesCrud],
+      },
+    },
+  });
+  const { id, deepLinks, ...rest } = casesLinks;
+  return {
+    ...rest,
+    id: SecurityPageName.case,
+    links: deepLinks.map((link) => {
+      const { id: subId, ...all } = link;
+      if (subId === SecurityPageName.caseConfigure) {
+        return { ...all, id: SecurityPageName.caseConfigure };
+      }
+      return { ...all, id: SecurityPageName.caseCreate };
```

Comment:
I love this function :D but I don't get why we need to do this id overriding, aren't they the same ids we are setting? if it is a type problem, couldn't we just do `links: deepLinks as LinkItem[]`?

Reply:
no, cases has its own page ids so we need to overwrite them

Reply:
true, but they have to be the same, otherwise navigation wouldn't work inside cases. 
I built that part, cases have a function to export the deepLinksIds to the solutions that needs it, it is used in observability, but in security the ids are defined in an enum (&>#%!) and it's not possible to add values from an external object, I had to hardcode them and add [this warning](https://github.com/elastic/kibana/blob/5ab0fa580d6b707e5aaa570acd25f7e57fc686b2/x-pack/plugins/security_solution/common/constants.ts#L81)

So yes, you can do the `as LinkItem[]`, they will never have different ids.